### PR TITLE
Arrays with proper prefix and namespace

### PR DESCRIPTION
**Pull request by vimalraj-a**
When Array has specific prefix and namespace as per the WSDL, the namespace reference is ignored completely. 

The current fix makes the array elements with prefix and namespace when available. (Not only during the first element and don't conflict with **_overrideRootElement_** option)

### CODE REVIEW

**Comment by jsdevel** [comment]
@vimalraj-a please add tests and ensure the build passes

**Comment by vimalraj-a** [comment]
@jsdevel I have added test case. Now CI is failing not sure what is the cause. Can you please review my changes?

File: test/client-test.js [hunk 1]
```diff
@@ -1578,6 +1578,49 @@ it('should create async client without options', function (done) {
     done();
   });
 });
+// console.log(__dirname)
```
Comment (jsdevel) [inline test/client-test.js:1581]: please remove this comment

**Review by jsdevel — CHANGES_REQUESTED** [review]

File: test/client-test.js (hunk 1 above)
Comment (vimalraj-a) [reply test/client-test.js]: Done

**Review by jsdevel — CHANGES_REQUESTED** [review]
The build is failing because of lint errors.  Please fix those and make the build pass.

**Comment by vimalraj-a** [comment]
> The build is failing because of lint errors. Please fix those and make the build pass.

Lint and test fixes are done. Please re-review the changes

**Comment by vimalraj-a** [comment]
@jsdevel Please review and merge this pull request. Let me know if anything needs change

Thanks,
Vimalraj

**Comment by vtrphan** [comment]
since 0.37.0 i'm having problem with prefixes too. For example i have an attribute Roles which is an array, like
Roles : [{ RoleName: "someName", RoleDesc: "a Description"}],
then i prefix it manually so that it becomes like:
"ns1:Roles": [{ "ns1:RoleName": "someName", "ns1:RoleDesc": "a Description"}],
then i pass this array into node-soap, the xml request it produces will be like
<Roles><ns1:RoleName>someName</ns1:RoleName><ns1:RoleDesc>a Description</ns1:RoleDesc></Roles>.
Notice the namespace "ns1" has been stripped off from "Roles". This caused my request to fail. Any idea or fix?
This behavior is since v0.37.0. Before that, all worked well.

**Comment by vimalraj-a** [comment]
Can you provide more details about the  WSDL? @vtrphan . If possible provide reproducible code.

**Comment by vtrphan** [comment]
here's the wsdl:
```
<?xml version="1.0" encoding="UTF-8"?>
<wsdl:definitions xmlns:ns="urn:services.pu-system:v1.0" xmlns:env="urn:wsdltypes.pu-system:v1.0" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:http="http://schemas.xmlsoap.org/wsdl/http/" xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:wsaw="http://www.w3.org/2006/05/addressing/wsdl" name="G100.wsdl" targetNamespace="urn:services.pu-system:v1.0">
	<wsdl:types>
		<xs:schema xmlns:types="urn:types.pu-system:v1.1" targetNamespace="urn:wsdltypes.pu-system:v1.0" xmlns="http://www.w3.org/2001/XMLSchema">
			<xs:import namespace="urn:types.pu-system:v1.1" schemaLocation="./v1/g400/G43032.xsd"/>
			<xs:import namespace="urn:types.pu-system:v1.1" schemaLocation="./v1/g700/G770.xsd"/>
			<xs:import namespace="urn:types.pu-system:v1.1" schemaLocation="./v1/g700/G775.xsd"/>
			<xs:import namespace="urn:types.pu-system:v1.1" schemaLocation="./v1/g400/G440.xsd"/>
			
			<xs:element name="G43032Request">
				<xs:complexType>
					<xs:complexContent>
						<xs:extension base="types:G43032Request_Type"/>
					</xs:complexContent>
				</xs:complexType>
			</xs:element>
			<xs:element name="G43032Response">
				<xs:complexType>
					<xs:complexContent>
						<xs:extension base="types:G43032Response_Type"/>
					</xs:complexContent>
				</xs:complexType>
			</xs:element>
			<xs:element name="G440Request">
				<xs:complexType>
					<xs:complexContent>
						<xs:extension base="types:G440Request_Type"/>
					</xs:complexContent>
				</xs:complexType>
			</xs:element>
			<xs:element name="G440Response">
				<xs:complexType>
					<xs:complexContent>
						<xs:extension base="types:G440Response_Type"/>
					</xs:complexContent>
				</xs:complexType>
			</xs:element>
			<xs:element name="G770Request">
				<xs:complexType>
					<xs:complexContent>
						<xs:extension base="types:G770Request_Type"/>
					</xs:complexContent>
				</xs:complexType>
			</xs:element>
			<xs:element name="G770Response">
				<xs:complexType>
					<xs:complexContent>
						<xs:extension base="types:G770Response_Type"/>
					</xs:complexContent>
				</xs:complexType>
			</xs:element>
			<xs:element name="G775Request">
				<xs:complexType>
					<xs:complexContent>
						<xs:extension base="types:G775Request_Type"/>
					</xs:complexContent>
				</xs:complexType>
			</xs:element>
			<xs:element name="G775Response">
				<xs:complexType>
					<xs:complexContent>
						<xs:extension base="types:G775Response_Type"/>
					</xs:complexContent>
				</xs:complexType>
			</xs:element>
		</xs:schema>
	</wsdl:types>
	<wsdl:message name="G43032RequestMessage">
		<wsdl:part name="parameters" element="env:G43032Request"/>
	</wsdl:message>
	<wsdl:message name="G43032ResponseMessage">
		<wsdl:part name="parameters" element="env:G43032Response"/>
	</wsdl:message>
	<wsdl:message name="G440RequestMessage">
		<wsdl:part name="parameters" element="env:G440Request"/>
	</wsdl:message>
	<wsdl:message name="G440ResponseMessage">
		<wsdl:part name="parameters" element="env:G440Response"/>
	</wsdl:message>
	<wsdl:message name="G770RequestMessage">
		<wsdl:part name="parameters" element="env:G770Request"/>
	</wsdl:message>
	<wsdl:message name="G770ResponseMessage">
		<wsdl:part name="parameters" element="env:G770Response"/>
	</wsdl:message>
	<wsdl:message name="G775RequestMessage">
		<wsdl:part name="parameters" element="env:G775Request"/>
	</wsdl:message>
	<wsdl:message name="G775ResponseMessage">
		<wsdl:part name="parameters" element="env:G775Response"/>
	</wsdl:message>
	
	<wsdl:portType name="IInternal_Services">
		<wsdl:operation name="G43032Operation">
			<wsdl:input message="ns:G43032RequestMessage" wsaw:Action="ns:G43032RequestMessage"/>
			<wsdl:output message="ns:G43032ResponseMessage" wsaw:Action="ns:G43032ResponseMessage"/>
		</wsdl:operation>
		<wsdl:operation name="G770Operation">
			<wsdl:input message="ns:G770RequestMessage" wsaw:Action="ns:G770RequestMessage"/>
			<wsdl:output message="ns:G770ResponseMessage" wsaw:Action="ns:G770ResponseMessage"/>
		</wsdl:operation>
		<wsdl:operation name="G775Operation">
			<wsdl:input message="ns:G775RequestMessage" wsaw:Action="ns:G775RequestMessage"/>
			<wsdl:output message="ns:G775ResponseMessage" wsaw:Action="ns:G775ResponseMessage"/>
		</wsdl:operation>
		<wsdl:operation name="G440Operation">
			<wsdl:input message="ns:G440RequestMessage" wsaw:Action="ns:G440RequestMessage"/>
			<wsdl:output message="ns:G440ResponseMessage" wsaw:Action="ns:G440ResponseMessage"/>
		</wsdl:operation>
	</wsdl:portType>
	
	<wsdl:binding name="WSHttpBinding_IInternal_Services" type="ns:IInternal_Services">
		<soap:binding style="document" transport="http://schemas.xmlsoap.org/soap/http"/>
		<wsdl:operation name="G43032Operation">
			<soap:operation soapAction="urn:G43032Operation" style="document"/>
			<wsdl:input>
				<soap:body use="literal"/>
			</wsdl:input>
			<wsdl:output>
				<soap:body use="literal"/>
			</wsdl:output>
		</wsdl:operation>
		<wsdl:operation name="G440Operation">
			<soap:operation soapAction="urn:G440Operation" style="document"/>
			<wsdl:input>
				<soap:body use="literal"/>
			</wsdl:input>
			<wsdl:output>
				<soap:body use="literal"/>
			</wsdl:output>
		</wsdl:operation>
		<wsdl:operation name="G770Operation">
			<soap:operation soapAction="urn:G770Operation" style="document"/>
			<wsdl:input>
				<soap:body use="literal"/>
			</wsdl:input>
			<wsdl:output>
				<soap:body use="literal"/>
			</wsdl:output>
		</wsdl:operation>
		<wsdl:operation name="G775Operation">
			<soap:operation soapAction="urn:G775Operation" style="document"/>
			<wsdl:input>
				<soap:body use="literal"/>
			</wsdl:input>
			<wsdl:output>
				<soap:body use="literal"/>
			</wsdl:output>
		</wsdl:operation>
	</wsdl:binding>
	<wsdl:service name="InternalServices">
		<wsdl:port name="Port_Internalt_Services" binding="ns:WSHttpBinding_IInternal_Services">
			<soap:address location="http://localhost:8080/WS_INTERNAL_TRANSACTIONS_V1/InternalServiceV10"/>
		</wsdl:port>
	</wsdl:service>
</wsdl:definitions>
```

**Comment by vimalraj-a** [comment]
Looks like xsd files are missing.

```
<xs:import namespace="urn:types.pu-system:v1.1" schemaLocation="./v1/g400/G43032.xsd"/>
			<xs:import namespace="urn:types.pu-system:v1.1" schemaLocation="./v1/g700/G770.xsd"/>
			<xs:import namespace="urn:types.pu-system:v1.1" schemaLocation="./v1/g700/G775.xsd"/>
			<xs:import namespace="urn:types.pu-system:v1.1" schemaLocation="./v1/g400/G440.xsd"/>
```
Create reproducible code in stackpath or provide code repo with samples for better understanding.

Create a bug or let's discuss it on Gitter.

**Comment by willemruys** [comment]
@vimalraj-a hi Vimalraj, I've opened a [pull request](https://github.com/vpulim/node-soap/pull/1184) with changes unrelated to the changes in this pull request, and it seems the test you've written in this pull request is failing. 

I cannot seem to find what changes made to fail these tests. Could you maybe have a look and see what is happening? Any help/info is much appreciated 😄 .

**Pull request merged**